### ISSUE TITLE
Overviews on image layers

### ISSUE THREAD

**Issue by djhoese** [ssec/sift#26]
In GitLab by @djhoese on Sep 25, 2015, 13:23

An overview is a very low resolution copy of the image being displayed that fits in to one tile. This image should always be shown when the image layer as a whole is visible. This way fast panning and zooming shows something to the user.

**Comment by djhoese** [comment]
In GitLab by @djhoese on Sep 25, 2015, 13:25

There are a couple choices for how to implement this:

 * A separate ImageVisual layer that has its own node in the scene graph.
    - Not the best idea. Super easy to implement, but separates a part of the image from the layer object
 * Add it as just another tile to the image layer but have special code to display it all the time.
    - This should be possible if and only if we can guarantee that the overview is written first so that other tiles are on top of it. If possible, this is probably the easiest thing to implement.
    - The other alternative to this would be to somehow specify the Z level of just that one large tile, but that isn't possible unless all vertices are done that way (I think, haven't researched).
 * Add it as another program, but still controlled by the image layer.
    - The biggest complication here is that we have to make sure everywhere the base ImageVisual class touches the original program or its buffers that we do similarly for the overview program.

**Comment by djhoese** [comment]
In GitLab by @djhoese on Oct 1, 2015, 10:24

I've completed this with commit 478c27e9a0a49c52087f944956157b0a72971316. I chose to do it with method 2 above. It sits in the texture and never expires. I was able to place the vertices and texture coordinates at the end of the arrays and that made the overview image get drawn behind all of the higher resolution tiles. This may be dependent on the specific GPU or version of some library, but for now it works.
